momento-sdk/src/main/java/momento/sdk/CacheClient.java

```diff
@@ -1199,6 +1201,134 @@ public CompletableFuture<CacheDictionarySetFieldResponse> dictionarySetField(
+   *
+   * <p>The Time to Live (TTL) seconds defaults to the parameter used when building this Cache
+   * client - {@link CacheClient#builder(CredentialProvider, Configuration, Duration)}
+   *
+   * @param cacheName - The cache containing the dictionary.
+   * @param dictionaryName - The dictionary to set the field in.
+   * @param items - The fields to set.
+   * @return Future containing the result of the dictionary fetch back operation.
+   */
+  public CompletableFuture<CacheDictionarySetFieldsResponse> dictionarySetFieldsStringString(
+      String cacheName, String dictionaryName, Map<String, String> items) {
+    return scsDataClient.dictionarySetFieldsStringString(cacheName, dictionaryName, items, null);
+  }
+
+  /**
+   * Sets all the fields in the given dictionary.
+   *
+   * @param cacheName - The cache containing the dictionary.
+   * @param dictionaryName - The dictionary to set the field in.
+   * @param items - The fields to set.
+   * @param ttl Time to Live for the item in Cache. This ttl takes precedence over the TTL used when
+   *     building a cache client {@link CacheClient#builder(CredentialProvider, Configuration,
+   *     Duration)}
+   * @return Future containing the result of the dictionary fetch back operation.
+   */
+  public CompletableFuture<CacheDictionarySetFieldsResponse> dictionarySetFieldsStringBytes(
+      String cacheName, String dictionaryName, Map<String, byte[]> items, CollectionTtl ttl) {
+    return scsDataClient.dictionarySetFieldsStringBytes(cacheName, dictionaryName, items, ttl);
+  }
+
+  /**
+   * Sets all the fields in the given dictionary.
+   *
+   * <p>The Time to Live (TTL) seconds defaults to the parameter used when building this Cache
+   * client - {@link CacheClient#builder(CredentialProvider, Configuration, Duration)}
+   *
+   * @param cacheName - The cache containing the dictionary.
+   * @param dictionaryName - The dictionary to set the field in.
+   * @param items - The fields to set.
+   * @return Future containing the result of the dictionary fetch back operation.
+   */
+  public CompletableFuture<CacheDictionarySetFieldsResponse> dictionarySetFieldsStringBytes(
+      String cacheName, String dictionaryName, Map<String, byte[]> items) {
+    return scsDataClient.dictionarySetFieldsStringBytes(cacheName, dictionaryName, items, null);
+  }
+
+  /**
+   * Sets all the fields in the given dictionary.
+   *
+   * @param cacheName - The cache containing the dictionary.
+   * @param dictionaryName - The dictionary to set the field in.
+   * @param items - The fields to set.
+   * @param ttl Time to Live for the item in Cache. This ttl takes precedence over the TTL used when
+   *     building a cache client {@link CacheClient#builder(CredentialProvider, Configuration,
+   *     Duration)}
+   * @return Future containing the result of the dictionary fetch back operation.
+   */
+  public CompletableFuture<CacheDictionarySetFieldsResponse> dictionarySetFieldsBytesString(
```

Comment:
The .NET sdk doesn't have a <byte[], String> implementation. I wonder if we need this. We can always remove it later.

Reply:
Yeah, I saw that. But, decided to follow the [API Spec doc](https://www.notion.so/momentohq/SDK-API-Methods-Spec-cf12e4db2535405ca2acc3608814b083?pvs=4#8f9824b9b86d482ab971eaae03d56eb3). The nodejs sdk implements all of them, so I agree to keep it for now and remove later if not required!